File: py/InsertionSort.py
from typing import List


class InsertionSort:
    def __init__(self, array):
        self.array = array

    def sort(self) -> List:
        """Sorts the array using Insertion Sort."""
        for i in range(1, len(self.array)):
            self.insert(i)
        return self.array

    def insert(self, i):
        """Inserts the 'Transition element' into its correct position in the sorted portion of the array."""
    #     TODO: TO BE IMPLEMENTED

    def swap(self, i, j):
        self.array[i], self.array[j] = self.array[j], self.array[i]

File: java/src/main/java/org/example/InsertionSort.java
package org.example;

public class InsertionSort {
    //Sorts the array using Insertion Sort.
    public void sort(Comparable[] a) {
        for (int i = 1; i < a.length; i++) {
            insert(i, a);
        }
    }

    private void insert(int i, Comparable[] a) {
        //Inserts the 'Transition element' into its correct position in the sorted portion of the array.
        //TODO: TO BE IMPLEMENTED
        for (int j = 0; j < i; j ++){
            if (a[i].compareTo(a[j]) > 0 ){
                continue;
            } else if (a[i].compareTo(a[j]) < 0){
                for (int y = i; y > j; y--){
                    swap(a, y,y-1);
                }
            }
        }
    }

    private void swap(Object[] a, int i, int j) {
        Object temp = a[i];
        a[i] = a[j];
        a[j] = temp;
    }
}
